lib/internal/fs/utils.js

```diff
@@ -907,8 +907,12 @@ const validateRmOptionsSync = hideStackFrames((path, options, expectDir) => {
   return options;
 });
 
-let recursiveRmdirWarned = process.noDeprecation;
+let recursiveRmdirWarned;
 function emitRecursiveRmdirWarning() {
+  if (recursiveRmdirWarned === undefined) {
+    // TODO(joyeecheung): use getOptionValue('--no-deprecation') instead.
+    recursiveRmdirWarned = process.noDeprecation;
```

Comment:
`recursiveRmdirWarned = recursiveRmdirWarned  || process.noDeprecation;`

Reply:
I think that would make the code harder to understand when we are lazily initializing a boolean.